File: fibonacci_race.m
#import <stdio.h>

int main(int argc, char *argv[]) {
    // TODO: #import <Foundation/Foundation.h>の読み込みができないため使えないクラスが多い…。
    printf("%d\n", fibonacci(19));
    printf("%d\n", fibonacci(9));
    printf("%d\n", fibonacci(2));
    printf("%d\n", fibonacci(16));
    printf("%d\n", fibonacci(3));
    printf("%d\n", fibonacci(8));
    printf("%d\n", fibonacci(0));
    printf("%d\n", fibonacci(6));
    printf("%d\n", fibonacci(4));
    printf("%d\n", fibonacci(17));
    printf("%d\n", fibonacci(5));
    printf("%d\n", fibonacci(1));
    printf("%d\n", fibonacci(14));
    printf("%d\n", fibonacci(12));
    printf("%d\n", fibonacci(15));
    printf("%d\n", fibonacci(13));
    printf("%d\n", fibonacci(10));
    printf("%d\n", fibonacci(7));
    printf("%d\n", fibonacci(11));
    printf("%d\n", fibonacci(18));
    return 0;
}

int fibonacci(int n) {
    if (n < 2) return n;
    return fibonacci(n - 2) + fibonacci(n - 1);
}

File: fibonacci_race.cpp
// fibonacci_cpp.cpp : コンソール アプリケーションのエントリ ポイントを定義します。
//

#include <stdio.h>

int fibonacci(int i)
{
	if(i <2)
	{
		return i;
	}
	else
	{
		return fibonacci(i-1) + fibonacci(i-2);
	}
}

int main(int argc, char* argv[])
{	
	int inputArray[] = {19,9,2,16,3,8,0,6,4,17,5,1,14,12,15,13,10,7,11,18};

	int inputLength = sizeof(inputArray) / sizeof(inputArray[0]);
	
	int i = 0;
	for(i =0; i<inputLength; i++)
	{
		int fiboNumber = fibonacci(inputArray[i]);
		printf("%d\n",fiboNumber);
	}	
}